platform/mellanox/docker-syncd-mlnx/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,466 @@
+
+def sx_get_active_vrids(handle):
+    """ Get existing virtual router IDs
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+    
+    Returns:
+        list : List of virtual routers ids
+    """      
+    try:
+        vrid_list = []
+        
+        vrid_cnt_p = new_uint32_t_p()
+        uint32_t_p_assign(vrid_cnt_p, 0)
+        vrid_key_p = new_sx_router_id_t_p()
+        sx_router_id_t_p_assign(vrid_key_p, 0)
+        vrid_key = sx_router_id_t_p_value(vrid_key_p)
+        
+        rc = sx_api_router_vrid_iter_get(handle, SX_ACCESS_CMD_GET, vrid_key, None, None, vrid_cnt_p)
+        sx_check_rc(rc)
+        
+        vrid_cnt = uint32_t_p_value(vrid_cnt_p)
+        vrid_list_p = new_sx_router_id_t_arr(vrid_cnt)
+        
+        rc = sx_api_router_vrid_iter_get(handle, SX_ACCESS_CMD_GET_FIRST, vrid_key, None, vrid_list_p, vrid_cnt_p)
+        sx_check_rc(rc)
+        
+        vrid_cnt = uint32_t_p_value(vrid_cnt_p)
+        for i in range(0, vrid_cnt):
+            vrid = sx_router_id_t_arr_getitem(vrid_list_p, i)
+            vrid_list.append(vrid)
+            
+        return vrid_list
+    
+    finally:
+        delete_uint32_t_p(vrid_cnt_p)
+        delete_sx_router_id_t_p(vrid_key_p)
+        delete_sx_router_id_t_arr(vrid_list_p)
+
+def sx_get_ports_map(handle):
+    """ Get ports map from SDK logical index to SONiC index
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+    
+    Returns:
+        dict : Dictionary of ports indices. Key is SDK logical index, value is SONiC index (4 for Ethernet4)
+    """         
+    try:
+        ports_map = {}
+        
+        # Get chip type
+        chip_type = sx_get_chip_type(handle)
+        
+        # Get ports count
+        port_cnt_p = new_uint32_t_p()
+        rc = sx_api_port_device_get(handle, DEVICE_ID, SWITCH_ID, None,  port_cnt_p)
+        sx_check_rc(rc)
+        
+        # Get ports
+        port_cnt = uint32_t_p_value(port_cnt_p)
+        port_attributes_list = new_sx_port_attributes_t_arr(port_cnt)
+        rc = sx_api_port_device_get(handle, DEVICE_ID, SWITCH_ID, port_attributes_list,  port_cnt_p)
+        sx_check_rc(rc)
+        
+        for i in range(0, port_cnt):
+            port_attributes = sx_port_attributes_t_arr_getitem(port_attributes_list, i)
+            label_port = port_attributes.port_mapping.module_port
+            logical_port = port_attributes.log_port;
+            lane_bmap = port_attributes.port_mapping.lane_bmap;
+            
+            if (is_phys_port(port_attributes.log_port) == False):
+                continue
+            
+            # Calculate sonic index (sonic index=4 for Ethernet4)
+            lane_index = get_lane_index(lane_bmap, ASIC_MAX_LANES[chip_type])
+            assert lane_index != -1, "Failed to calculate port index"
+            
+            sonic_index = label_port * ASIC_MAX_LANES[chip_type] + lane_index;
+            sonic_interface = ETHERNET_PREFIX + str(sonic_index)    
```

Comment:
This is why we have it in define ETHERNET_PREFIX. If change is needed we just change the define. 
Also, the define now moved to a lib so it can be used by others.